src/Microsoft.TestPlatform.CommunicationUtilities/JsonDataSerializer.cs

```diff
@@ -121,30 +125,41 @@ public string SerializeMessage(string messageType)
         /// <returns>Serialized message.</returns>
         public string SerializePayload(string messageType, object payload)
         {
-            JToken serializedPayload = null;
-
-            // TODO: Currently we use json serializer auto only for non-testmessage types
-            // CHECK: Can't we just use auto for everything
-            if (MessageType.TestMessage.Equals(messageType))
-            {
-                serializedPayload = JToken.FromObject(payload);
-            }
-            else
-            {
-                serializedPayload = JToken.FromObject(payload, serializer);
-            }
+            var serializedPayload = JToken.FromObject(payload, payloadSerializer);
```

Comment:
We might want to change the APIs.
This is used for payload serialization/deserialization. Currently, the APIs caller reads the version beforehand